org.civicrm.bootstrapcivicrm/scss/civicrm/vacancies/_vacancies.scss

```diff
@@ -114,4 +114,42 @@
     margin: 0 !important;
     min-height: 35px !important;
   }
+
+  .CRM_HRRecruitment_Form_HRVacancy {
+    tbody:first-child {
+      tr:nth-child(8),
```

Comment:
As for javascript part, you're targeting a specific type of element only by using the parent's position. Isn't there something in the markup to identify fields for which you need to add the icon?